### PR TITLE
Added cal-vin example

### PR DESCRIPTION
**Pull request by milo157**

### CODE REVIEW

File: v4/examples/langchain_langsmith.mdx [hunk 1]
```diff
@@ -0,0 +1,428 @@
...
+```
+[cerebrium.dependencies.pip]
+pydantic = "latest"
+langchain = "latest"
+pytz = "latest" ##this is used for timezones
+openai = "latest"
+langchain_openai = "latest"
+```
+
+We will be using OpenAI GPT3.5 for our use cases and so we need an API key from them. If you don’t have an account, you can sign up [here](https://openai.com/). You can then create an API key [here](https://platform.openai.com/settings/profile?tab=api-keys). The API key should be in the format: “sk_xxxxx”.
```
Comment (kylegani) [inline v4/examples/langchain_langsmith.mdx:154]: User keys have been deprecated in favour of project keys. May want to update that
```suggestion
We will be using OpenAI GPT3.5 for our use cases and so we need an API key from them. If you don’t have an account, you can sign up [here](https://openai.com/). You can then create an API key [here](https://platform.openai.com/api-keys). The API key should be in the format: “sk_xxxxx”.
```

File: v4/examples/langchain_langsmith.mdx [hunk 2]
```diff
@@ -0,0 +1,428 @@
...
+  ‍
+
+Below is the code to achieve this
+
+```python
+
+from langchain_core.tools import tool
+from cerebrium import get_secret
+import requests
+from cal import find_available_slots
```
Comment (kylegani) [inline v4/examples/langchain_langsmith.mdx:177]: You don't actually show the user how to do this in the example (They have to navigate to the repo to find out). This is disconcerting as you only figure this out when you actually try to execute the main python file locally much lower down in the example

File: v4/examples/langchain_langsmith.mdx [hunk 3]
```diff
@@ -0,0 +1,428 @@
...
+        print(result)
+```
+
+The above code does the following:
+
+- We define a Pydantic object which specifies the parameters our API expects - the user prompt and a session id to tie the conversation to.
+- The predict function in Cerebrium is the entry point for our API so we just pass the prompt and session id to our agent and print the results.
+  ‍
+
+To run this, simply run the following in your terminal: python main.py. You will need to replace your secrets with the actual values when running locally. You should then see output similar to the following:
```
Comment (kylegani) [inline v4/examples/langchain_langsmith.mdx:325]: May want to also include that users should manually install pip packages added to toml if they don't have those dependencies. PS: You also need langchain-community to run this. See error below 
```ModuleNotFoundError: Module langchain_community.chat_message_histories not found. Please install langchain-community to access this module. You can install it using `pip install -U langchain-community` ```

```suggestion
To run this, simply install the pip dependencies manually by typing the following into your terminal `pip install pydantic langchain pytz openai langchain_openai langchain-community` and then run `python main.py` to execute your main python file. You will need to replace your secrets with the actual values when running locally. You should then see output similar to the following:
```

File: v4/examples/langchain_langsmith.mdx [hunk 4]
```diff
@@ -0,0 +1,428 @@
...
+- Get availability tool
+- You would have seen from the test API request we did above to Cal.com that the API returns your availability in the following way:
+- The time slots that you are already busy
+- Your working hours on each day
+  ‍
+
+Below is the code to achieve this
+
+```python
+
```
Comment (kylegani) [inline v4/examples/langchain_langsmith.mdx:173]: May want to include `from pydantic import BaseModel` here. It doesn't come included in recent init changes

**Review by kylegani — CHANGES_REQUESTED** [review]

File: v4/examples/langchain_langsmith.mdx [hunk 5]
```diff
@@ -0,0 +1,428 @@
...
+        "description": description,
+    }
+    response = requests.post(url, params=params)
+    if response.status_code == 200:
+        booking_data = response.json()
+        return booking_data
+    else:
+        print('error')
+        print(response)
+        return {}
```
Comment (kylegani) [inline v4/examples/langchain_langsmith.mdx:236]: This worked for me:
```python
def book_slot(datetime: str, name: str, email: str, title: str, description: str) -> float:
    """Book a meeting on my calendar at the requested date and time using the 'datetime' variable. Get a description about what the meeting is about and make a title for it"""
    url = "https://api.cal.com/v1/bookings"
    params = {
        "apiKey": get_secret("CAL_API_KEY"),
    }
    data = {
        "eventTypeId": 742028,
        "start": datetime,
        "responses": {
            "name": name,
            "email": email,
            "guests": [],
            "location": {
              "value": "inPerson",
              "optionValue": ""
            }
        },
        "timeZone": "Europe/London",
        "language": "en",
        "status": "PENDING",
        "title": title,
        "metadata": {},
        "description": description,
    }

    response = requests.post(url, params=params, json=data)
    if response.status_code == 200:
        booking_data = response.json()
        return booking_data
    else:
        print('error')
        print(response.json())
        return {}
```

**Review by kylegani — CHANGES_REQUESTED** [review]
Managed to get things working. Please see code that worked for booking. Also, when the booking fails, you still get a message to say that the booking was made successfully. Please see screenshot.
Screenshot 2024-05-30 at 12 47 22

File: v4/examples/langchain_langsmith.mdx (hunk 2 above)
Comment (milo157) [reply v4/examples/langchain_langsmith.mdx:177]: I explain it just below that

**Pull request merged**